login.js

```diff
@@ -0,0 +1,125 @@
+  } else {
+    removeError(password, currentError);
+  }
+
+  btnDisable();
+}
+
+function validateNickname() {
+  const nicknameValue = nickname.value.trim();
+  const currentError = document.querySelector(".nickname-error-message");
+
+  if (nicknameValue === "") {
+    showError(nickname, "닉네임을 입력해주세요.", "nickname-error-message");
+  } else {
+    removeError(nickname, currentError);
+  }
+
+  btnDisable();
+}
+
+function validateSignupConfirm() {
+  const signupConfirmValue = signupConfirm.value.trim();
+  const currentError = document.querySelector(".confirm-error-message");
+
+  if (signupConfirmValue !== password.value) {
+    showError(signupConfirm, "비밀번호가 일치하지 않습니다", "confirm-error-message");
+  } else {
+    removeError(signupConfirm, currentError);
+  }
+
+  btnDisable();
+}
+
+//에러 메세지
+function showError(input, message, errorClass) {
+  input.classList.add("error");
+
+  let errorElement = document.querySelector(`.${errorClass}`);
+  if (!errorElement) {
+    errorElement = document.createElement("span");
+    errorElement.classList.add(errorClass);
+    input.insertAdjacentElement("afterend", errorElement);
+  }
+  errorElement.textContent = message;
+}
+
+//에러 메세지 삭제
+function removeError(input, errorElement) {
+  input.classList.remove("error");
+  if (errorElement) {
+    errorElement.remove();
+  }
+}
+
+// 버튼 활성화/비활성화 함수
+function btnDisable() {
+  const emptyInput = Array.from(inputs).some((input) => input.value.trim() === "");
+  const error =
+    document.querySelector(".email-error-message") ||
+    document.querySelector(".pw-error-message") ||
+    document.querySelector(".nickname-error-message") ||
+    document.querySelector(".confirm-error-message");
+
+  if (emptyInput || error) {
+    submit.disabled = true;
+    submit.classList.add("disabled");
+  } else {
+    submit.disabled = false;
+    submit.classList.remove("disabled");
+  }
+}
+
+email.addEventListener("focusout", validateEmail);
+password.addEventListener("focusout", validatePassword);
+if (nickname) {
+  nickname.addEventListener("focusout", validateNickname);
+}
+if (signupConfirm) {
+  signupConfirm.addEventListener("focusout", validateSignupConfirm);
+}
+
+//심화
+const visibilityBtn = document.querySelector(".btn_visibility");
+const visibilityBtnCheck = document.querySelector(".check");
+
+visibilityBtn.addEventListener("click", function () {
+  const visibilityImg = visibilityBtn.querySelector("img");
+  password.type = password.type === "text" ? "password" : "text";
+  visibilityImg.src = password.type === "text" ? "../images/btn_visibility_on.png" : "../images/btn_visibility_off.png";
+});
+visibilityBtnCheck.addEventListener("click", function () {
+  const visibilityImgCheck = visibilityBtnCheck.querySelector("img");
+  signupConfirm.type = signupConfirm.type === "text" ? "password" : "text";
+  visibilityImgCheck.src = signupConfirm.type === "text" ? "../images/btn_visibility_on.png" : "../images/btn_visibility_off.png";
+});
```

Comment:
이렇게 해주면 이벤트리스너를 각각 DOM조회해서 두번씩 사용할필요없이 코드가 더 명료해질수있겠죠? 그리고 접근성을 위해 사용했던 aria-pressed 속성도 꼭 업데이트해주시고요!

```suggestion
const visibilityBtns = document.querySelectorAll(".btn_visibility");

visibilityBtns.forEach(btn => {
  btn.addEventListener("click", function() {
    const input = this.parentElement.querySelector("input");
    const img = this.querySelector("img");
    
    input.type = input.type === "text" ? "password" : "text";
    img.src = input.type === "text" ? "../images/btn_visibility_on.png" : "../images/btn_visibility_off.png";
    this.setAttribute("aria-pressed", input.type === "text");
  });
});
```